editor/src/app/sites/shared/file-upload.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { take, map, catchError } from 'rxjs/operators';
+import { Store } from '@ngxs/store';
+import { AppShowLoading, AppHideLoading } from '../../app-state/app.actions';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FileUploadService {
+
+  constructor(
+    private store: Store,
+    private http: HttpClient) {
+  }
+
+  upload(property, file) {
+    const url = '/engine/upload.php?property=' + property;
+    const formData = new FormData();
+    formData.append('Filedata', file);
+
+    this.store.dispatch(new AppShowLoading());
+
+    return this.http.post(url, formData).pipe(
```

Comment:
You can specify the response type when doing request: `this.http.post<{filename: string}>(url, formData)`. This way it will be easier to use the response later.